posthog/hogql_queries/web_analytics/web_active_hours_heatmap_query_runner.py

```diff
@@ -27,28 +27,20 @@ def __init__(self, *args, **kwargs):
     def to_query(self) -> ast.SelectQuery:
         query = parse_select(
             """
-            WITH (
-                SELECT
-                    uniqMap(map(concat(toString(toDayOfWeek(timestamp)), ',' ,toString(toHour(timestamp))), events.person_id)) as hoursAndDays,
-                    uniqMap(map(toHour(timestamp), events.person_id)) as hours,
-                    uniqMap(map(toDayOfWeek(timestamp), events.person_id)) as days,
-                    uniq(person_id) as total
-                FROM events
-                WHERE and(
-                    event = '$pageview',
-                    {all_properties},
-                    {current_period}
-                )
-            ) as cte
             SELECT
-                mapKeys(cte.hoursAndDays) as hoursAndDaysKeys,
-                mapValues(cte.hoursAndDays) as hoursAndDaysValues,
-                mapKeys(cte.hours) as hoursKeys,
-                mapValues(cte.hours) as hoursValues,
-                mapKeys(cte.days) as daysKeys,
-                mapValues(cte.days) as daysValues,
-                cte.total
-            FROM cte
+                mapKeys(uniqMap(map(concat(toString(toDayOfWeek(timestamp)), ',' ,toString(toHour(timestamp))), events.person_id))) as hoursAndDaysKeys,
+                mapValues(uniqMap(map(concat(toString(toDayOfWeek(timestamp)), ',' ,toString(toHour(timestamp))), events.person_id))) as hoursAndDaysValues,
+                mapKeys(uniqMap(map(toHour(timestamp), events.person_id))) as hoursKeys,
+                mapValues(uniqMap(map(toHour(timestamp), events.person_id))) as hoursValues,
+                mapKeys(uniqMap(map(toDayOfWeek(timestamp), events.person_id))) as daysKeys,
+                mapValues(uniqMap(map(toDayOfWeek(timestamp), events.person_id))) as daysValues,
```

Comment:
Ahn, I actually think you should've kept the concept of a subquery because you're now computing the `hoursAndDays` map twice - and `hours` and `days`

My proposal was:
- before: `WITH [...] AS query SELECT ... FROM query`
- after: `SELECT ... from [...] AS query`

Reply:
Yeah, that's exactly it, thanks!